tests/beta/test_default_qubit_tf.py

```diff
@@ -1045,174 +1020,195 @@ def circuit(a, b):
+                [
+                    np.sin(theta) * np.cos(lam) * np.cos(phi),
+                    np.cos(theta) * np.cos(lam) * np.sin(phi) + np.sin(lam) * np.cos(phi),
+                    np.cos(theta) * np.sin(lam) * np.cos(phi) + np.cos(lam) * np.sin(phi),
+                ]
+            )
+            * 2
+            * (np.sin(lam) * np.sin(phi) - np.cos(theta) * np.cos(lam) * np.cos(phi))
+        )
+        assert np.allclose(res.numpy(), expected_grad, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("interface", ["autograd", "torch"])
+    def test_error_backprop_wrong_interface(self, interface, tol):
+        """Tests that an error is raised if diff_method='backprop' but not using
+        the TF interface"""
         dev = qml.device("default.qubit.tf", wires=1)
 
-        @qml.qnode(dev, diff_method="backprop", interface=interface)
-        def circuit(x, weights, w=None):
-            """In this example, a mixture of scalar
-            arguments, array arguments, and keyword arguments are used."""
-            qml.QubitStateVector(1j * np.array([1, -1]) / np.sqrt(2), wires=w)
-            # the parameterized gate is one that gets decomposed
-            # via a template
-            qml.U3.decomposition(x, weights[0], weights[1], wires=w)  # <--- decomposition is used
+        def circuit(x, w=None):
+            qml.RZ(x, wires=w)
             return qml.expval(qml.PauliX(w))
 
-        def cost_fn(params):
-            """Perform some classical processing"""
-            return circuit(params[0], params[1:], w=0) ** 2
+        with pytest.raises(
+            ValueError,
+            match="default.qubit.tf only supports the tf interface when diff_method='backprop'",
+        ):
+            qml.qnode(dev, diff_method="backprop", interface=interface)(circuit)
 
-        return cost_fn
 
-    @pytest.mark.parametrize("interface", ["tf"])
-    def test_tf_interface_classical_diff(self, cost_with_decomposition, interface, tol):
-        """Tests that the gradient of an arbitrary U3 gate (that gets
-        decomposed) is correct using the TensorFlow interface and the classical
-        diff method"""
-        # TODO: once the decomposition of operations and the PassThruQNode are
-        # compatible, merge this case into the previous one
-        import tensorflow as tf
+class TestSamplesNonAnalytic:
+    """Tests for sampling and non-analytic mode"""
 
-        params = tf.Variable(self.p, dtype=tf.float64)
+    def test_sample_observables(self):
+        """Test that the device allows for sampling from observables."""
+        shots = 100
+        dev = qml.device("default.qubit.tf", wires=2, shots=shots)
+
+        @qml.qnode(dev, diff_method="backprop", interface="tf")
+        def circuit(a):
+            qml.RX(a, wires=0)
+            return qml.sample(qml.PauliZ(0))
+
+        a = tf.Variable(0.54)
+        res = circuit(a)
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (1, shots)

```

Comment:
:thinking:  I know I keep losing this argument, but I don't like having these singleton dimensions appearing unnecessarily

Reply:
This surprised me as well while I was testing. I initially expected the shape to just be `(shots,)`.